### DISCUSSION TITLE
Add support for Hue Play Gradient Lightstrip for PC

### DISCUSSION THREAD

**Discussion by hmmSalat** [vchlum/hue-lights#57] [Ideas]
Hey, 

I just bought the new Hue Play Gradient Lightstrip for PC. Apparently the display sync feature isn't fully supported by the Hue Lights extension. 

https://www.philips-hue.com/en-gb/p/lightstrips-hue-play-gradient-lightstrip-fur-pc/8719514434479

Is it possible to add support for this?

**Comment by vchlum** [comment]
Hi! It should be possible. I will probably need some help because I do not own the Gradient Lightstrip for PC... Are you willing to help me? I will probably need some data from you. Also, help with testing would be useful.

**Reply by hmmSalat** [reply]
Yes, of course. What kind of data do you need?

**Reply by vchlum** [reply]
OK:-) Please, open `dconf-editor` and go to the path `/org/gnome/shell/extensions/hue-lights/bridges`. Here, get the `<ip>` and `<username>` of your bridge.
Knowing this, run the following command locally:
`wget http://<ip>/api/<username>/` and provide me the file index.html.

It is a JSON file containing the state of your bridge. It does not contain sensitive data. If you want to be extra sure, feel free to remove the key 'config' from the JSON. I believe the most sensitive attribute is the MAC address of your device... You can also send it via email instead of providing it publicly.

**Reply by hmmSalat** [reply]
You've got mail ;-)

**Reply by vchlum** [reply]
Thank you. Please, test this branch: https://github.com/vchlum/hue-lights/tree/gradientstrippc Seems like an easy fix. Let me know if all the segments of the gradient strip for pc work. If not, I will need API2 data from your bridge:-)

**Reply by hmmSalat** [reply]
Wow, that was quick! First test says it seems to be working now. Thank you very much! Will do some more testing later.

**Reply by hmmSalat** [reply]
Hey! Here you can see what it looks like: https://youtu.be/gsXlhCbbYss
On the left side you can see the sync of the official windows hue sync app. On the right side you can see your extension doing the work.
It's difficult to see in the video but there is one difference: The official hue sync app splits the light strip into 5 segments. The hue lights extension for gnome splits it into 7 segments. I think this causes some weird flickering of the light strip when using the hue lights gnome extension. Maybe the flickering could be fixed by decreasing the number of segments from 7 to 5. The part of the light strip which is on top of the screen should be handled as one segment. See this sketch: https://ibb.co/tpYFrxw

**Reply by vchlum** [reply]
IMO, the 5 segments are available in the app for control. In reality, the gradient light strip has 7 segments available for the entertainment API. A theoretical 5 segments gradient light strip would shift the synchronized lights... There would be actually "top" lights on the right side of your display - simply shifted... but your screen is synchronized correctly from left to right. The flickering is an issue in the algorithm. It was introduced with a new gnome version. I will investigate and I will try to fix it. Since I am a little bit short on time, I just can not say the ETA.

**Reply by hmmSalat** [reply]
Okay, thank you for the information. Take your time. Please let me know if you need any more data.
